ecoli/analysis/multigeneration/ribosome_crowding.py

```diff
@@ -0,0 +1,250 @@
+        gene_id = next(
+            (
+                g
+                for c, g in zip(mRNA_data["id"], mRNA_data["gene_id"])
+                if c == cistron_id
+            ),
+            "Unknown",
+        )
+        monomer_to_gene[mono_id] = gene_id
+
+    # Get field metadata
+    try:
+        field_names = field_metadata(
+            conn,
+            config_sql,
+            "listeners__ribosome_data__target_prob_translation_per_transcript",
+        )
+    except Exception as e:
+        print(f"[ERROR] Error getting field metadata: {e}")
+        return
+
+    # First pass: Find overcrowded monomer indices
+    overcrowded_query = f"""
+    WITH unnested AS (
+        SELECT 
+            unnest(listeners__ribosome_data__actual_prob_translation_per_transcript) as actual,
+            unnest(listeners__ribosome_data__target_prob_translation_per_transcript) as target,
+            generate_subscripts(listeners__ribosome_data__target_prob_translation_per_transcript, 1) as idx
+        FROM ({history_sql})
+    )
+    SELECT DISTINCT idx
+    FROM unnested
+    WHERE target > actual
+    ORDER BY idx
+    LIMIT {MAX_NUMBER_OF_MONOMERS_TO_PLOT}
+    """
+
+    overcrowded_indices = [
+        row[0] - 1 for row in conn.execute(overcrowded_query).fetchall()
+    ]  # Convert to 0-based
+
+    if not overcrowded_indices:
+        print("[INFO] No overcrowded monomers found.")
+        return
+
+    n_overcrowded_monomers = len(overcrowded_indices)
+    n_overcrowded_monomers_to_plot = min(
+        n_overcrowded_monomers, MAX_NUMBER_OF_MONOMERS_TO_PLOT
+    )
+
+    print(f"[INFO] Found {n_overcrowded_monomers} overcrowded monomers")
+
+    # Second pass: Get data for overcrowded monomers only
+    actual_columns = []
+    target_columns = []
+
+    for i, idx in enumerate(overcrowded_indices):
+        if i >= n_overcrowded_monomers_to_plot:
+            break
+        if idx < len(field_names):
+            monomer_id = field_names[idx]
+            gene_id = monomer_to_gene.get(monomer_id, "Unknown")
+            actual_columns.append(f"actual__{gene_id}")
+            target_columns.append(f"target__{gene_id}")
+
+    actual_expr = named_idx(
+        "listeners__ribosome_data__actual_prob_translation_per_transcript",
+        actual_columns,
+        [overcrowded_indices[: len(actual_columns)]],
+    )
+
+    target_expr = named_idx(
+        "listeners__ribosome_data__target_prob_translation_per_transcript",
+        target_columns,
+        [overcrowded_indices[: len(target_columns)]],
+    )
+
+    data_query = f"SELECT {actual_expr}, {target_expr}, time FROM ({history_sql})"
+    df = conn.execute(data_query).fetchdf()
+
+    # ----------------------------------------- #
+    # Prepare plot data following original format
+    pl_df = pl.DataFrame(df)
+
+    # Get all probability columns (both actual and target)
+    prob_columns = actual_columns + target_columns
```

Comment:
The ordering of these columns causes the `actual` rows to appear before the `target` rows after unpivoted. Altair will then draw the orange dashed line (`actual`) first before the blue solid line (`target`). I honestly like the look of this plot better because it more clearly shows where the two are different. If anything, I'd probably just remove the different stroke types below.